lib/ansible/plugins/action/command.py

```diff
@@ -17,7 +17,7 @@ def run(self, tmp=None, task_vars=None):
         del tmp  # tmp no longer has any effect
 
         # Command module has a special config option to turn off the command nanny warnings
-        if 'warn' not in self._task.args:
+        if 'warn' not in self._task.args and C.COMMAND_WARNINGS:
```

Comment:
this changes the meaning, before we only set the configured default if it was not set in the task itself, now we only set it if the configured default is true.

what you really want is to set it if the configured default is explicit and not from 'default', so those setting it to 'false' in the configuration should also get a warning. (use `get_config_value_and_origin` to get both value and where it was set from)